File: 공부내용정리/알고리즘/leetcode/HashMap/Two Sum/TwoSum.ts
function twoSum(nums: number[], target: number): number[] {
  const map = new Map<number, number>();

  for( let i = 0; i < nums.length; i++) {
      const otherNum = target - nums[i];

      if(map.has(otherNum)) {
          return [map.get(otherNum)!, i];
      }

    map.set(nums[i], i);
  }

  return []

};

File: 공부내용정리/알고리즘/leetcode/HashMap/Two Sum/TwoSum.java
import java.util.Map;
import java.util.HashMap;

class Solution {
  public int[] twoSum(int[] nums, int target) {
    Map<Integer, Integer> map = new HashMap<>();

    for ( int i = 0; i < nums.length; i++ ) {
      int otherNum = target - nums[i];

      if (map.containsKey(otherNum)) {
          return new int[] { map.get(otherNum), i};
      }

      map.put(nums[i], i);
    }  

    return new int[0];
  }
}